### PR TITLE
feat: Add Invite storage

### LINKED ISSUE #458: Invites permanent storage
Add database persistence for invites

### PR DESCRIPTION
**Pull request by Elyniss**
Closes #458 

Merge after #630 
Next in line #635

### CODE REVIEW

**Review by jakubuid — APPROVED** [review]

File: chat/sdk/src/main/kotlin/com/walletconnect/chat/common/model/InviteStatus.kt [hunk 1]
```diff
@@ -1,5 +1,4 @@
-@file:JvmSynthetic
-
 package com.walletconnect.chat.common.model
 
-internal enum class InviteStatus { PENDING, REJECTED, APPROVED }
\ No newline at end of file
+// Used in Invites.sq
+enum class InviteStatus { PENDING, REJECTED, APPROVED }
```
Comment (TalhaAli00) [inline chat/sdk/src/main/kotlin/com/walletconnect/chat/common/model/InviteStatus.kt:4]: why does this not need to be internal?

File: chat/sdk/src/main/kotlin/com/walletconnect/chat/common/model/InviteType.kt [hunk 2]
```diff
@@ -0,0 +1,4 @@
+package com.walletconnect.chat.common.model
+
+// Used in Invites.sq
+enum class InviteType { SENT, RECEIVED }
```
Comment (TalhaAli00) [inline chat/sdk/src/main/kotlin/com/walletconnect/chat/common/model/InviteType.kt:4]: same

File: chat/sdk/src/main/kotlin/com/walletconnect/chat/storage/InvitesStorageRepository.kt [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+            inviterPublicKey = invite.inviterPublicKey, inviteePublicKey = invite.inviteePublicKey
+        )
+    }
+    fun deleteInviteByInviteId(inviteId: Long) = invites.deleteInviteByInviteId(inviteId)
+
+    fun updateStatusByInviteId(inviteId: Long, status: InviteStatus) = invites.updateInviteStatus(status, inviteId)
+
+    fun getSentInvitesForInviterAccount(inviterAccount: String): List<Invite.Sent> = invites.getSentInvitesForInviterAccount(inviterAccount, ::dbToSentInvite).executeAsList()
+
+    fun getReceivedInvitesForInviteeAccount(inviterAccount: String): List<Invite.Received> = invites.getReceivedInvitesForInviteeAccount(inviterAccount, ::dbToReceivedInvite).executeAsList()
```
Comment (TalhaAli00) [inline chat/sdk/src/main/kotlin/com/walletconnect/chat/storage/InvitesStorageRepository.kt:27]: can we make these suspend functions?

**Review by TalhaAli00 — APPROVED** [review]

File: chat/sdk/src/main/kotlin/com/walletconnect/chat/common/model/InviteStatus.kt (hunk 1 above)
Comment (Elyniss) [reply chat/sdk/src/main/kotlin/com/walletconnect/chat/common/model/InviteStatus.kt:4]: I had issues with generated code. It didn't had access to this class

File: chat/sdk/src/main/kotlin/com/walletconnect/chat/common/model/InviteType.kt (hunk 2 above)
Comment (Elyniss) [reply chat/sdk/src/main/kotlin/com/walletconnect/chat/common/model/InviteType.kt:4]: I had issues with generated code. It didn't have access to this class

File: chat/sdk/src/main/kotlin/com/walletconnect/chat/common/model/InviteStatus.kt (hunk 1 above)
Comment (Elyniss) [reply chat/sdk/src/main/kotlin/com/walletconnect/chat/common/model/InviteStatus.kt:4]: Screenshot 2023-02-15 at 09 26 02

File: chat/sdk/src/main/kotlin/com/walletconnect/chat/storage/InvitesStorageRepository.kt (hunk 3 above)
Comment (Elyniss) [reply chat/sdk/src/main/kotlin/com/walletconnect/chat/storage/InvitesStorageRepository.kt]: Made them suspend, as well as, some engine function that only calls them. Blocking is done inside ChatProtocol class

**Pull request merged**